### PR TITLE
Build support for the TypeScript target

### PR DESCRIPTION
**Pull request by lhstrh**
- [x] address Clippy warnings
- [x] port logic for selecting `pnpm` if it exists, and falling back on `npm` if it doesn't

### CODE REVIEW

**Review by revol-xut — COMMENTED** [review]
- First I would actually split it into two backend npm and pnpm this would clean up this significantly because then all the weird `if` go
- Then for testing the cmake backend drops all the build files right now into a `target` directory similar to cargo 
- Otherwise this already gives a structure which now just needs to be thoroughly test

File: defaults/HelloTS.lf
Comment (lhstrh) [inline defaults/HelloTS.lf:1]: This file is not properly formatted. Run `lff` on it?

**Review by lhstrh — COMMENTED** [review]
The latest changes look great, @revol-xut!

File: defaults/HelloTS.lf
Comment (revol-xut) [reply defaults/HelloTS.lf:1]: done

**Comment by lhstrh** [comment]
Shall we merge this?

**Comment by revol-xut** [comment]
I think its ready

**Review by revol-xut — APPROVED** [review]

**Pull request merged**